### ISSUE TITLE
Clear token cache on any error

### ISSUE THREAD

**Issue by databus23** [sapcc/concourse-swift-resource#1]
Currently the check resource might get stuck on a bad cached token, re-using it indefinitely.
The cache should be cleared on any errors to avoid deadlocks.

**Comment by JohannesRudolph** [comment]
Absolutely... considering that some implementations of Swift (most noteworthy Ceph radosgw) have `http://tracker.ceph.com/issues/21226#change-99128` on the backend and it's an absolute PITA to debug cache invalidation errors.
